internal/local/chart/kube.go

```diff
@@ -111,3 +112,83 @@ func watchWarningEvents(ctx context.Context, namespace string, kubeconfig []byte
 
 	return ch, watcher.Stop, nil
 }
+
+// GetNonRuninngPods returns a list of non-running pods in the ReleaseNamespace namespace
+func GetNonRuninngPods() ([]corev1.Pod, error) {
+	kubeconfig, err := ReadKubeConfig(KubeConfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := clientcmd.RESTConfigFromKubeConfig(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	pods, err := clientset.CoreV1().Pods(ReleaseNamespace).List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var nonRunningOrCompletedPods []corev1.Pod
+	for _, pod := range pods.Items {
+		for _, containerStatus := range pod.Status.ContainerStatuses {
+			state := containerStatus.State
+			if state.Waiting != nil || state.Terminated != nil {
+				if state.Terminated != nil && state.Terminated.Reason == "Completed" {
+					continue
+				}
+				nonRunningOrCompletedPods = append(nonRunningOrCompletedPods, pod)
+				break
```

Comment:
why break ? dont you wanna run on all items and collect all non running pods ?
or will this break only the internal loop?